content/chainguard/chainguard-images/getting-started/nemo/index.md

```diff
@@ -0,0 +1,3013 @@
+
+NeMo is built on [PyTorch Lightning](https://lightning.ai/docs/pytorch/stable/), a high-level interface to PyTorch with a focus on scalability, and uses the [Hydra](https://hydra.cc/) library for configuration management.
+
+Since NeMo is a framework with many collections of modules suitable for a wide variety of projects, we've chosen an example task, generative text to speech, requiring the use of two TTS modules. This is an appropriate example of a task that might be run as part of a larger production application.
+
+## Text to Speech (TTS) Example
+
+In this section, we'll run a script that uses the NeMo Chainguard Image to:
+
+- Start with a message in plain text
+- Transform it into a set of phonemes
+- Generate a spectrogram (waveform representation) using a NeMo-provided spectrogram model
+- Transform the spectrogram values into audio using a NeMo-provided vocoder (human voice) model
+- Write the resulting audio to a `.wav` file at a set rate
+
+First, let's create a folder to work in on your host machine:
+
+```bash
+mkdir -p ~/nemo-tts && cd ~/nemo-tts
+```
+
+Next, let's download our [tts.py](https://github.com/chainguard-dev/nemo-examples/blob/main/tts.py) script:
+
+```bash
+curl https://raw.githubusercontent.com/chainguard-dev/nemo-examples/main/tts.py > tts.py
+```
+
+You should now be in a working directory with `tts.py`.
+
+```bash
+$ ls
+tts.py
+```
+
+We'll be mounting this folder in our container as a volume, which will allow us to both pass in our script and extract our output.
+
+We'll now start a container based on our NeMo Chainguard Image, mount the current working directory containing our `tts.py` script inside the container as a volume, and run the script in the container:
+
+```bash
+$ docker run -it --rm \
+  --gpus all \
+  --user root \
+  --shm-size=8g \
+  --ulimit memlock=-1 \
+ --ulimit stack=67108864 \
+  -v $PWD:/home/nonroot/nemo-test \
+  cgr.dev/chainguard/nemo:latest \
+  -c "python /home/nonroot/nemo-test/tts.py"
+```
+
+If your host machine does not have attached GPUs and you'd like to run the above on your CPU, omit the `  --gpus all \` line. Since we're using pretrained models to perform text to speech, this example will only take a few minutes using a CPU only. However, other tasks such as model training and finetuning may take significantly longer without connected GPUs.
+
+Note that NeMo collections are large, and initial imports can take up to a minute depending on your environment. The script may appear to hang during that time.
+
+After imports are complete, you should see a large amount of output as NeMo pulls models and works through the steps in the script (tokenizing, generating a spectrogram, generating audio, and writing audio to disk). On completion, the script outputs a `test.wav` file. Because we mounted a volume, this file should now be present in the working directory of your host machine.
+
+```bash
+$ ls
```

Comment:
This can look a bit confusing with the command and output together in the same block. I don't think you need to add the `ls` command tbh. If you want to, it's best to have them in separate blocks to optimized rendering:

```shell
ls
```

```
output
```

Reply:
I removed one `ls` command and left the other, as I feel like it shows the final state more clearly than another paragraph, etc. I hear you on the $, I removed them from the docker commands as they do make copying a pain. We might want to talk as a team about options for including prompts in a non-copyable way on CG Academy, something like a smart shortcode or similar.